vignettes/mmrm_review_methods.Rmd

```diff
@@ -1043,3 +911,177 @@ rel_diff_ests_tbl %>%
+        data = miss_df[complete.cases(miss_df), ],
+        correlation = nlme::corSymm(form = ~ VISITN | USUBJID),
+        weights = nlme::varIdent(form = ~ 1|VISITN)
+      )
+      gls_converged <- is.null(gls_fit$error)
+
+      ## glmmTMB
+      glmmTMB_fit <- glmmTMB(
+        BCVA_CHG ~ BCVA_BL + RACE + ARMCD * AVISIT + us(0 + AVISIT | USUBJID),
+        data = miss_df,
+        dispformula = ~ 0
+      )
+      glmmTMB_converged <- glmmTMB_fit$fit$convergence == 0
+      
+      ## SAS
+      if (file.exists("sascfg_personal.py")) {
+
+        # compute marginal estimates from PROC MIXED
+        # load the data into SAS
+        df2sd(miss_df, "miss_df")
+
+        sas_code <- "ODS OUTPUT ConvergenceStatus = conv_status;
+          PROC MIXED DATA = miss_df cl method=reml;
+            CLASS AVISIT(ref = 'VIS01') RACE(ref = 'Asian') ARMCD(ref = 'CTL') USUBJID;
+            MODEL BCVA_CHG = BCVA_BL ARMCD AVISIT ARMCD*AVISIT RACE / ddfm=satterthwaite solution chisq;
+            REPEATED AVISIT / subject=USUBJID type=un r rcorr;
+          RUN;
+        "
+
+        # generate the SAS output
+        sas_result <- run_sas(sas_code)
+
+        ## extract the convergence status
+        conv_status_df <- sasr::sd2df("conv_status")
+        sas_converged <- conv_status_df$Reason == "Convergence criteria met."
+
+        ## assemble tibble row
+        results <- tibble(
+          replicate = idx,
+          method = c("mmrm", "lmer", "gls", "glmmTMB", "PROC MIXED"),
+          converged = c(mmrm_converged, lmer_converged, gls_converged,
+                        glmmTMB_converged, sas_converged)
+        )
+        
+      } else {
+        results <- tibble(
+          replicate =  idx,
+          method = c("mmrm", "lmer", "gls", "glmmTMB"),
+          converged = c(mmrm_converged, lmer_converged, gls_converged,
+                        glmmTMB_converged)
+        )
+      }
+
+      return(results)
+    }
+  ) %>%
+  group_by(method) %>%
+  summarize(convergence_rate = mean(converged), .groups = "drop") %>%
+  mutate(missingness = missingness_level)
+}
+
+## get the convergence rates
+n_reps <- 10
+cr_no_missing <- get_convergence_rates(n_reps, "none")
+cr_mild_missing <- get_convergence_rates(n_reps, "mild")
+cr_moderate_missing <- get_convergence_rates(n_reps, "moderate")
+cr_high_missing <- get_convergence_rates(n_reps, "high")
+cr_all <- bind_rows(
+  cr_no_missing, cr_mild_missing, cr_moderate_missing, cr_high_missing
+)
+
+## plot the convergence rates
+cr_all %>%
+  mutate(
+    missingness = factor(
+      missingness,
+      levels = c("none", "mild", "moderate", "high")
+    )
+  ) %>%
+  ggplot(aes(x = method, y = convergence_rate)) +
+  geom_point() +
+  facet_grid(rows = vars(missingness)) +
+  xlab("Method") +
+  ylab("Convergence Rate (10 Replicates)") +
+  ggtitle("Convergence Rates by Missingness Levels") +
+  scale_y_continuous(labels = scales::percent_format(accuracy = 1)) +
+  theme_bw()
+```
+
```

Comment:
Can you please add a high-level summary / interpretation what can be seen in the plot?